homeassistant/components/rituals_perfume_genie/switch.py

```diff
@@ -0,0 +1,95 @@
+"""Support for Rituals Perfume Genie switches."""
+from datetime import timedelta
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+ON_STATE = "1"
+AVAILABLE_STATE = 1
+
+MANUFACTURER = "Rituals Cosmetics"
+MODEL = "Diffuser"
+ICON = "mdi:fan"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the diffuser switch."""
+    session = async_get_clientsession(hass)
+    account = hass.data[DOMAIN][config_entry.entry_id]
+    diffusers = await account.get_devices(session)
+
+    entities = []
+    for diffuser in diffusers:
+        entities.append(DiffuserSwitch(diffuser, session))
+
+    async_add_entities(entities, True)
+
+
+class DiffuserSwitch(SwitchEntity):
+    """Representation of a diffuser switch."""
+
+    def __init__(self, diffuser, session):
+        """Initialize the switch."""
+        self._diffuser = diffuser
+        self._session = session
+
+    @property
+    def device_info(self):
+        """Return information about the device."""
+        return {
+            "name": self._diffuser.data["hub"]["attributes"]["roomnamec"],
+            "identifiers": {(DOMAIN, self._diffuser.data["hub"]["hublot"])},
+            "manufacturer": MANUFACTURER,
+            "model": MODEL,
+            "sw_version": self._diffuser.data["hub"]["sensors"]["versionc"],
+        }
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the device."""
+        return self._diffuser.data["hub"]["hublot"]
+
+    @property
+    def available(self):
+        """Return if the device is available."""
+        return self._diffuser.data["hub"]["status"] == AVAILABLE_STATE
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._diffuser.data["hub"]["attributes"]["roomnamec"]
+
+    @property
+    def icon(self):
+        """Return the icon of the device."""
+        return ICON
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attributes = {
+            "fan_speed": self._diffuser.data["hub"]["attributes"]["speedc"],
```

Comment:
Does it make more sense to implement this as a fan since it has a fan speed?

https://developers.home-assistant.io/docs/core/entity/fan

Reply:
There is a small fan to blow the perfume out of the unit. It is not visible and you cannot feel the wind. The user does not know that there is a fan in it (unless he opens the unit to change the perfume).
You can set the "perfume amount" with the app, in the API it is called "speedc" and in reality it is the speed of the fan. I chose "fan_speed" instead of "perfume_amount" to prevent confusion with the amount of perfume that is left. In the app, it is clearly an setting. But in HASS it is unclear whether it is a setting or a measurement.
The user wants to turn on the perfume distribution and not a fan.